redis/mutation_utils.cc

```diff
@@ -0,0 +1,93 @@
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "redis/mutation_utils.hh"
+#include "types.hh"
+#include "service/storage_service.hh"
+#include "schema.hh"
+#include "database.hh"
+#include <seastar/core/print.hh>
+#include "redis/keyspace_utils.hh"
+#include "redis/options.hh"
+#include "mutation.hh"
+#include "service_permit.hh"
+
+using namespace seastar;
+
+namespace redis {
+
+atomic_cell make_cell(const schema_ptr schema,
+        const abstract_type& type,
+        bytes_view value,
+        long cttl = 0)
+{
+
+    if (cttl > 0) {
+        auto ttl = std::chrono::seconds(cttl);
+        return atomic_cell::make_live(type, api::new_timestamp(), value, gc_clock::now() + ttl, ttl, atomic_cell::collection_member::no);
+    }   
+    auto ttl = schema->default_time_to_live();
+    if (ttl.count() > 0) {
+        return atomic_cell::make_live(type, api::new_timestamp(), value, gc_clock::now() + ttl, ttl, atomic_cell::collection_member::no);
+    }   
+    return atomic_cell::make_live(type, api::new_timestamp(), value, atomic_cell::collection_member::no);
+}  
+
+mutation make_mutation(service::storage_proxy& proxy, const redis_options& options, bytes&& key, bytes&& data, long ttl) {
+    auto schema = get_schema(proxy, options.get_keyspace_name(), redis::STRINGs);
+    const column_definition& column = *schema->get_column_definition(redis::DATA_COLUMN_NAME);
+    auto pkey = partition_key::from_single_value(*schema, key);
+    auto m = mutation(schema, std::move(pkey));
+    auto cell = make_cell(schema, *(column.type.get()), data, ttl);
+    m.set_clustered_cell(clustering_key::make_empty(), column, std::move(cell));
+    return std::move(m);
+}
+
+future<> write_strings(service::storage_proxy& proxy, redis::redis_options& options, bytes&& key, bytes&& data, long ttl, service_permit permit) {
+    db::timeout_clock::time_point timeout = db::timeout_clock::now() + options.get_write_timeout();
+    auto m = make_mutation(proxy, options, std::move(key), std::move(data), ttl);
+    auto write_consistency_level = options.get_write_consistency_level();
+    return proxy.mutate(std::vector<mutation> {std::move(m)}, write_consistency_level, timeout, nullptr, permit);
+}
+
+
+mutation make_tombstone(service::storage_proxy& proxy, const redis_options& options, const sstring& cf_name, const bytes& key) {
+    auto schema = get_schema(proxy, options.get_keyspace_name(), cf_name);
+    auto pkey = partition_key::from_single_value(*schema, key);
+    auto m = mutation(schema, std::move(pkey));
+    m.partition().apply(tombstone { api::new_timestamp(), gc_clock::now() }); 
+    return std::move(m);
```

Comment:
```
FAILED: build/release/redis/mutation_utils.o 
g++ -MD -MT build/release/redis/mutation_utils.o -MF build/release/redis/mutation_utils.o.d -I/home/amos/scylladb.com/scylla-build-example/scylla-redis/seastar/include -I/home/amos/scylladb.com/scylla-build-example/scylla-redis/build/release/seastar/gen/include -std=gnu++17 -U_FORTIFY_SOURCE -Wno-maybe-uninitialized -march=westmere -Werror=unused-result -fconcepts -DSEASTAR_API_LEVEL=2 -DSEASTAR_HAVE_GCC6_CONCEPTS -DSEASTAR_USE_STD_OPTIONAL_VARIANT_STRINGVIEW -DSEASTAR_HAVE_DPDK -DFMT_SHARED -I/usr/include/p11-kit-1 -ffile-prefix-map=/home/amos/scylladb.com/scylla-build-example/scylla-redis=.  -march=westmere -DBOOST_TEST_DYN_LINK -DTHRIFT_USES_BOOST  -fvisibility=hidden -Wall -Werror -Wno-maybe-uninitialized -Wno-tautological-compare -Wno-missing-braces -Wno-misleading-indentation -Wno-overflow -Wno-noexcept-type -Wno-nonnull-compare -Wno-error=cpp -Wno-ignored-attributes -Wno-overloaded-virtual -Wno-stringop-overflow -Wno-error=deprecated-declarations -DXXH_PRIVATE_API -DSEASTAR_TESTING_MAIN -DHAVE_LIBSYSTEMD=1 -DHAVE_LZ4_COMPRESS_DEFAULT -O3 --param inline-unit-growth=300 -gz  -g -I. -I build/release/gen  -c -o build/release/redis/mutation_utils.o redis/mutation_utils.cc
redis/mutation_utils.cc: In function 'mutation redis::make_mutation(service::storage_proxy&, const redis::redis_options&, bytes&&, bytes&&, long int)':
redis/mutation_utils.cc:61:21: error: moving a local object in a return statement prevents copy elision [-Werror=pessimizing-move]
   61 |     return std::move(m);
      |            ~~~~~~~~~^~~
redis/mutation_utils.cc:61:21: note: remove 'std::move' call
redis/mutation_utils.cc: In function 'mutation redis::make_tombstone(service::storage_proxy&, const redis::redis_options&, const sstring&, const bytes&)':
redis/mutation_utils.cc:77:21: error: moving a local object in a return statement prevents copy elision [-Werror=pessimizing-move]
   77 |     return std::move(m);
      |            ~~~~~~~~~^~~
redis/mutation_utils.cc:77:21: note: remove 'std::move' call
cc1plus: all warnings being treated as errors
```

Reply:
Thanks. The std::move is unnecessary. I'll fix it.